src/read/macinfo.rs

```diff
@@ -0,0 +1,290 @@
+                    file_number_index,
+                })
+            }
+            constants::DW_MACINFO_end_file => {
+                // no operands
+                Some(DebugMacInfoItem::EndFile)
+            }
+            constants::DW_MACINFO_vendor_ext => {
+                // two operands: a constant (LEB128) and a null terminated string, whose meaning is vendor specific
+                let numeric_constant = self.input_reader.read_uleb128().ok()?;
+                let string_constant = self.input_reader.read_null_terminated_slice().ok()?;
+                Some(DebugMacInfoItem::VendorExt {
+                    numeric_constant,
+                    string_constant,
+                })
+            }
+            _ => {
+                // unknown macinfo type; return None, since Err is not an option here
+                self.finished = true;
+                return None;
+            }
+        }
+    }
+}
+
+impl<R: Reader> FusedIterator for DebugMacInfoIterator<R> {}
+
+#[cfg(test)]
+mod tests {
+    use crate::{test_util::GimliSectionMethods, LittleEndian};
+
+    use super::*;
+    use test_assembler::{Endian, Label, LabelMaker, Section};
+
+    #[test]
+    fn test_get_macinfo() {
+        let position = Label::new();
+
+        // Create a test section with some macinfo entries
+        let section = Section::with_endian(Endian::Little)
+            .set_start_const(0)
+            .mark(&position)
+            .D8(crate::DW_MACINFO_define.0)
+            .uleb(0) // line number: 0 - defined on the compiler command line
+            .append_bytes(b"__STDC__ 1\0")
+            .D8(crate::DW_MACINFO_define.0)
+            .uleb(1) // line number: 1 - defined in the source file
+            .append_bytes(b"__GNUC__ 1\0")
+            .D8(crate::DW_MACINFO_undef.0)
+            .uleb(2) // line number: 2 - undefined in the source file
+            .append_bytes(b"__GNUC__\0")
+            .D8(crate::DW_MACINFO_start_file.0)
+            .uleb(3) // line number: 3 - start of file
+            .uleb(4) // file number index: 4 - index into the line number table
+            .D8(crate::DW_MACINFO_end_file.0) // end of file
+            .D8(crate::DW_MACINFO_vendor_ext.0)
+            .uleb(5) // numeric constant: 5 - vendor specific
+            .append_bytes(b"foo\0")
+            .D8(crate::DW_MACINFO_end_of_unit.0); // end of unit
+
+        // Create a DebugMacInfo instance from the section
+        let section = section.get_contents().unwrap();
+        let debug_macinfo = DebugMacInfo::from(EndianSlice::new(&section, LittleEndian));
+
+        let offset = position.value().unwrap() as usize;
+
+        let mut iter = debug_macinfo.get_macinfo(DebugMacinfoOffset(offset));
+
+        // Test getting macinfo entries
+        let entry = iter.next().unwrap();
+        assert!(
+            matches!(entry, DebugMacInfoItem::Define { line_number, macro_text } if line_number == 0 && macro_text.to_string_lossy() == "__STDC__ 1")
```

Comment:
```suggestion
            matches!(entry, DebugMacInfoItem::Define { line_number: 0, macro_text } if macro_text.slice() == b"__STDC__ 1")
```
Similarly for others.